### ISSUE TITLE
Make it possible to delete past dev builds

### ISSUE THREAD

**Issue by ogrisel** [ogrisel/wheelhouse-uploader#13]
Some projects like numpy use a git digest information in the local segment of the version number. This makes it impossible for pip to know which is the most recent dev build wheel to install. Furthermore accumulating a large number of dev wheels is often useless and wastes storage resources.

We should therefore implement a scheme to delete past dev wheels (with matching platform tag and python version) by default after uploading a new dev wheel to the wheelhouse.

**Comment by ogrisel** [comment]
To clarify, here are example dev wheel filenames for numpy:

```
- numpy-1.11.0.dev0+53e658f-cp35-cp35m-linux_x86_64.whl
- numpy-1.11.0.dev0+d94043f-cp35-cp35m-linux_x86_64.whl
```

**Comment by ogrisel** [comment]
This is required for numpy/numpy#6493

**Comment by ogrisel** [comment]
Now merged and released in 0.8.0
